### DISCUSSION TITLE
External filter function provided to the table

### DISCUSSION THREAD

**Discussion by snowytime** [TanStack/table#3815] [Q&A] [answered]
Hey so im exploring react table, and want to be able to pass a function to the table that will filter it based on the data of `row.original`.

Is there a way to implement that using the `useFilters` hook?

example of a custom (external filter)

`const projectFilter = useCallback(
    rows => {
      if (project === 'all') return rows
      return rows.filter(r => r.original.project === project)
    },
    [project]
  )`

and then the table will be filtered accordingly, and respond to the `active` state change externally.

I've been following the docs but nothing seems to work for the use case.

Thx

**Comment by tannerlinsley** [comment, marked as answer]
You could easily use a column filter to do this, or a global filter, or you could even manually filter it with `useMemo` before passing it to the table.
